yt_dlp/extractor/zingmp3.py

```diff
@@ -244,13 +280,36 @@ class ZingMp3ChartHomeIE(ZingMp3BaseIE):
             'id': 'moi-phat-hanh',
         },
         'playlist_mincount': 100,
+    }, {
+        'url': 'https://zingmp3.vn/top100',
+        'info_dict': {
+            'id': 'top100',
+        },
+        'playlist_mincount': 50,
+    }, {
+        'url': 'https://zingmp3.vn/podcast-discover',
+        'info_dict': {
+            'id': 'podcast-discover',
+        },
+        'playlist_mincount': 150,
     }]
-    IE_NAME = 'zingmp3:chart-home'
 
     def _real_extract(self, url):
         url_type = self._match_id(url)
-        data = self._call_api(url_type, {'id': url_type})
-        items = traverse_obj(data, ('RTChart', 'items') if url_type == 'zing-chart' else 'items')
+        params = {'id': url_type}
+        if url_type == 'podcast-discover':
+            params.update({'type': 'discover'})
+        data = self._call_api(url_type, params)
+        items = []
+        if url_type == 'top100':
+            self.IE_NAME = 'zingmp3:top100'
+            items.extend(traverse_obj(data, (lambda _, v: v['items'], 'items', ...)))
+        elif url_type == 'podcast-discover':
+            self.IE_NAME = 'zingmp3:postcast-discover'
+            items.extend(traverse_obj(data, 'items'))
+        else:
+            self.IE_NAME = 'zingmp3:chart-home'
+            items.extend(traverse_obj(data, ('RTChart', 'items') if url_type == 'zing-chart' else 'items'))
```

Comment:
Pretty sure `IE_NAME` should not be changed during runtime.
As the `podcast-discover` case is the same as the default case this can be simplified to:
```suggestion
        if url_type == 'top100':
            items.extend(traverse_obj(data, (..., 'items', ...)))
        elif url_type == 'zing-chart':
            items.extend(traverse_obj(data, ('RTChart', 'items')))
        else:
            items.extend(traverse_obj(data, 'items'))
```

Reply:
Yeah, we should not be changing the `IE_NAME`.

Also, if we are using `extend`, we should always branch (e.g. with `...` or a filter) at some point in the traversal path to ensure it returns a list. And looks like we want a list of dicts, so let's expect the type, too (while expecting the type isn't really necessary since `_parse_items` uses traversal, I think it helps readability):
```suggestion
        if url_type == 'top100':
            items.extend(traverse_obj(data, (..., 'items', ..., {dict})))
        elif url_type == 'zing-chart':
            items.extend(traverse_obj(data, ('RTChart', 'items', ..., {dict})))
        else:
            items.extend(traverse_obj(data, ('items', ..., {dict})))
```